resources-generator/src/main/kotlin/dev/icerock/gradle/MultiplatformResourcesPlugin.kt

```diff
@@ -41,41 +44,84 @@ import org.jetbrains.kotlin.gradle.targets.js.ir.KotlinJsIrTarget
 import org.jetbrains.kotlin.gradle.targets.jvm.KotlinJvmTarget
 import org.jetbrains.kotlin.konan.target.HostManager
 import java.io.File
+import javax.inject.Inject
 import javax.xml.parsers.DocumentBuilderFactory
+import org.jetbrains.kotlin.gradle.dsl.KotlinProjectExtension
+import org.jetbrains.kotlin.gradle.plugin.extraProperties
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeCompilation
+import org.jetbrains.kotlin.gradle.plugin.mpp.pm20.util.targets
+import org.jetbrains.kotlin.gradle.targets.native.tasks.artifact.kotlinArtifactsExtension
+import org.jetbrains.kotlin.gradle.utils.ObservableSet
 
 @Suppress("TooManyFunctions")
-class MultiplatformResourcesPlugin : Plugin<Project> {
+abstract class MultiplatformResourcesPlugin : Plugin<Project> {
+    @Inject
+    protected abstract fun getObjectFactory(): ObjectFactory
+
     override fun apply(target: Project) {
         val mrExtension: MultiplatformResourcesPluginExtension = target.extensions.create(
             "multiplatformResources",
             MultiplatformResourcesPluginExtension::class
         )
         mrExtension.multiplatformResourcesPackage = "${target.group}.${target.name}"
 
-        target.plugins.withType(KotlinMultiplatformPluginWrapper::class) {
-            val multiplatformExtension =
-                target.extensions.getByType(KotlinMultiplatformExtension::class)
+        target.plugins.withType(KotlinMultiplatformPluginWrapper::class) { _ ->
+            val multiplatformExtension = target.extensions.getByType(
+                type = KotlinMultiplatformExtension::class
+            )
 
-            target.afterEvaluate {
-                configureGenerators(
-                    target = target,
-                    mrExtension = mrExtension,
-                    multiplatformExtension = multiplatformExtension
-                )
+            target.kotlinExtension.targets.forEach {kotlinTarget ->
+                kotlinTarget.compilations.configureEach { compilation ->
+                    compilation.allKotlinSourceSetsObservable.whenObjectAdded {
```

Comment:
whenObjectAdded does not guarantee at least one call to configureGenerators.
It is necessary to add a condition
```
if (compilation.allKotlinSourceSets.isEmpty().not()) {
                         configureGenerators(...)
```